halpybot/server/auth.py

```diff
@@ -34,45 +34,41 @@ def get_hmac(msg: str) -> hmac.HMAC:
     )
 
 
-def authenticate():
+def authenticate(function):
     """Validate a response coming in and check it against our config"""
 
-    def decorator(function):
-        @functools.wraps(function)
-        async def guarded(request):
-            # Add Connection Logger
-            connection_logger = logger.bind(task="API")
+    @functools.wraps(function)
+    async def guarded(request):
+        # Add Connection Logger
+        connection_logger = logger.bind(task="API")
```

Comment:
possible alternative to avoid needing to bind `logger` to a new name
```py
with logger.contextualize(task="API"):
   ... # calls made to logger in this context (or its children) will have the contextual information attached akin bind
```